src/server/game/Chat/Channels/Channel.cpp

```diff
@@ -288,7 +306,8 @@ void Channel::KickOrBan(Player const* player, std::string const& badname, bool b
         return;
     }
 
-    if (!playersStore[good].IsModerator() && !player->GetSession()->HasPermission(rbac::RBAC_PERM_CHANGE_CHANNEL_NOT_MODERATOR))
+    PlayerInfo& info = PlayersStore.at(good);
```

Comment:
if (!IsOn(good)) check handles this case and stops execution prematurely before reaching this line.